### PR TITLE
Prepare 1.28.0: step 2

### PR DESCRIPTION
**Pull request by djc**
Step 2 in our [release process](https://rust-lang.github.io/rustup/dev-guide/release-process.html#making-a-release):

- [x] Update the commit shasum in `rustup-init.sh` to match the latest commit on `master`.

### CODE REVIEW

File: rustup-init.sh
Comment (ChrisDenton) [inline rustup-init.sh]: Would it not be better that the release commit include the full changelog? Even though it has no effect on the binary itself.

**Review by ChrisDenton — APPROVED** [review]
Looks good to me. I have one query but I don't want to block on it.

File: rustup-init.sh
Comment (rami3l) [reply rustup-init.sh]: @chrisdenton Ideally we'd be able to do that, but not just yet. The plan is that this will be a placeholder for the release team to fill in. For now we just want the traceability (and since we're preparing for the beta, even this can be broken as long as it gets fixed in the final release).

**Review by rami3l — APPROVED** [review]
@djc Thanks a lot!

BTW are you planning to cut a beta release this week?

**Comment by djc** [comment]
> @djc Thanks a lot!
> 
> BTW are you planning to cut a beta release this week?

I plan to keep working towards it, don't have any kind of deadline in mind.

File: rustup-init.sh
Comment (djc) [reply rustup-init.sh]: Also given that this is just the beta release, I don't think it is as important for right now.

Comment (rami3l) [inline rustup-init.sh]: @djc Sorry for the interruption: you're breaking the output format by truncating the hash by one more digit :o

**Review by rami3l — APPROVED** [review]

**Comment by rami3l** [comment]
@djc These CI errors have also influenced https://github.com/rust-lang/rustup/pull/4070#issuecomment-2500886187.

```console
#5 0.935 E: The repository 'http://archive.ubuntu.com/ubuntu lunar-updates Release' no longer has a Release file.
```

`lunar` means Ubuntu 23.04 which have, I believe, already been EOL for quite a while now. Not sure why our [upstream](https://github.com/rust-lang/rust/blob/39cb3386ddc6c71657418be28dbb3987eea4aa4b/src/ci/docker/host-x86_64/arm-android/Dockerfile#L1) still hasn't updated it...

**Comment by rami3l** [comment]
> @djc These CI errors have also influenced [#4070 (comment)](https://github.com/rust-lang/rustup/pull/4070#issuecomment-2500886187).
> 
> ```
> #5 0.935 E: The repository 'http://archive.ubuntu.com/ubuntu lunar-updates Release' no longer has a Release file.
> ```
> 
> `lunar` means Ubuntu 23.04 which have, I believe, already been EOL for quite a while now. Not sure why our [upstream](https://github.com/rust-lang/rust/blob/39cb3386ddc6c71657418be28dbb3987eea4aa4b/src/ci/docker/host-x86_64/arm-android/Dockerfile#L1) still hasn't updated it...

It took quite a while, but my https://github.com/rust-lang/rust/pull/133532 has finally been merged! I guess we can restart the release process now.

**Comment by rami3l** [comment]
Oh, it's quite unfortunate since we're getting a Rust version bump in the middle of this PR, so we're introducing unresolved lints to the codebase...

**Pull request merged**
